docs/mds_test_plan.md

```diff
@@ -0,0 +1,39 @@
+# Metadata Service (mds) test plan
+
+### One-liner overview of the service
+The MDS is a blobstore used to store JSON-formatted data, any schemaless data.
+### RESTful API
+Here's one of its swagger pages:
+https://nci-crdc-staging.datacommons.io/mds/docs
+### Authentication
+Users can interact with MDS through basic auth, e.g.:
+```
+% curl --location --request POST 'https://${GEN3_HOSTNAME}/mds/metadata/someGUID2' \
+--header 'Content-Type: application/json' \
+--header "Authorization: Basic ${BASIC_AUTH_MDS}" \
+--data-raw '{
+        "bar": 789
+}'
+{"bar":789}
+```
+Or with a Gen3 Access Token, e.g.:
+```
+% curl -X POST 'https://${GEN3_HOSTNAME}/mds-admin/metadata/foo' \
+--header 'Content-Type: application/json' \
+--header "Authorization: Bearer ${ACCESS_TOKEN}" \
+--data-raw '{ "foo": 123 }'
+{"foo":123}
+```
+### Tests
+The initial coverage comprises the following scenarios:
+1. Create records against the metadata svc blobstore - positive test
+2. Get existing keys from metadata records - positive test
+3. Query metadata for matching records - positive test
+### Load Tests
+#### Performance benchmarking
+The service is currently being load tested with the following criteria:
+    http_req_duration: ['avg<3000', 'p(95)<15000'],
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [[no-undefined-references] Found reference to undefined definition](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)